framework/quota/test/org/apache/cloudstack/quota/QuotaAlertManagerImplTest.java

```diff
@@ -0,0 +1,205 @@
+        Field f = QuotaAlertManagerImpl.class.getDeclaredField(fieldName);
+        f.setAccessible(true);
+        f.set(quotaAlertManager, mock);
+    }
+
+    @Before
+    public void setup() throws IllegalAccessException, NoSuchFieldException, ConfigurationException {
+        // Dummy transaction stack setup
+        TransactionLegacy.open("QuotaAlertManagerImplTest");
+
+        injectMockToField(accountDao, "_accountDao");
+        injectMockToField(quotaAcc, "_quotaAcc");
+        injectMockToField(userDao, "_userDao");
+        injectMockToField(domainDao, "_domainDao");
+        injectMockToField(quotaEmailTemplateDao, "_quotaEmailTemplateDao");
+        injectMockToField(configDao, "_configDao");
+        injectMockToField(quotaUsage, "_quotaUsage");
+        injectMockToField(emailQuotaAlert, "_emailQuotaAlert");
+    }
+
+    @Test
+    public void testStartStop() {
+        try {
+            quotaAlertManager.start(); // expected to fail as pid is not available
+        } catch (NumberFormatException ignored) {
+        }
+        assertTrue(quotaAlertManager.stop());
+    }
+
+    @Test
+    public void testCheckAndSendQuotaAlertEmails() {
+        AccountVO accountVO = new AccountVO();
+        accountVO.setId(2L);
+        accountVO.setDomainId(1L);
+        accountVO.setType(Account.ACCOUNT_TYPE_NORMAL);
+        Mockito.when(accountDao.findById(Mockito.anyLong())).thenReturn(accountVO);
+
+        QuotaAccountVO acc = new QuotaAccountVO(2L);
+        acc.setQuotaBalance(new BigDecimal(404));
+        acc.setQuotaMinBalance(new BigDecimal(100));
+        acc.setQuotaBalanceDate(new Date());
+        acc.setQuotaAlertDate(null);
+        acc.setQuotaEnforce(0);
+        List<QuotaAccountVO> accounts = new ArrayList<>();
+        accounts.add(acc);
+        Mockito.when(quotaAcc.listAllQuotaAccount()).thenReturn(accounts);
+
+        // Don't test sendQuotaAlert yet
+        Mockito.doNothing().when(quotaAlertManager).sendQuotaAlert(Mockito.any(QuotaAlertManagerImpl.DeferredQuotaEmail.class));
+        Mockito.doReturn(true).when(quotaAlertManager).lockAccount(Mockito.anyLong());
+
+        // call real method on send monthly statement
+        Mockito.doCallRealMethod().when(quotaAlertManager).checkAndSendQuotaAlertEmails();
+
+        // Case1: valid balance, no email should be sent
+        quotaAlertManager.checkAndSendQuotaAlertEmails();
+        Mockito.verify(quotaAlertManager, Mockito.times(0)).sendQuotaAlert(Mockito.any(QuotaAlertManagerImpl.DeferredQuotaEmail.class));
+
+        // Case2: low balance, email should be sent
+        accounts.get(0).setQuotaBalance(new BigDecimal(99));
+        //Mockito.when(quotaAcc.listAll()).thenReturn(accounts);
+        quotaAlertManager.checkAndSendQuotaAlertEmails();
+        Mockito.verify(quotaAlertManager, Mockito.times(1)).sendQuotaAlert(Mockito.any(QuotaAlertManagerImpl.DeferredQuotaEmail.class));
```

Comment:
This test method has no asserts.  How do we verify the correctness of sending quota email as opposed no exception being thrown?